ui/src/app/utils/runtime-utils.ts

```diff
@@ -0,0 +1,93 @@
+import {runtimeApi} from 'app/services/swagger-fetch-clients';
+import {useOnMount} from 'app/utils';
+import {switchCase} from 'app/utils';
+import {
+  LeoRuntimeInitializer,
+} from 'app/utils/leo-runtime-initializer';
+import {
+  runtimeStore,
+  useStore
+} from 'app/utils/stores';
+import {Runtime, RuntimeStatus} from 'generated/fetch';
+import * as fp from 'lodash/fp';
+
+import * as React from 'react';
+
+const {useState, useEffect} = React;
+
+export type RuntimeStatusRequest = 'Delete';
+
+export const RuntimeStateRequest = {
+  Delete: 'Delete' as RuntimeStatusRequest
+};
+
+// useRuntime hook is a simple hook to populate the runtime store.
+// This is only used by other runtime hooks
+const useRuntime = (currentWorkspaceNamespace) => {
+  useOnMount(() => {
+    const getRuntime = async() => {
+      const leoRuntime = await runtimeApi().getRuntime(currentWorkspaceNamespace);
+      runtimeStore.set({
+        workspaceNamespace: currentWorkspaceNamespace,
+        runtime: leoRuntime
+      });
+    };
+    getRuntime();
+  });
+};
+
+// useRuntimeState hook can be used to change the state of the runtime
+// Only 'Delete' is supported at the moment
+export const useRuntimeState = (currentWorkspaceNamespace): [RuntimeStatus, Function]  => {
```

Comment:
This hook can be used to change the state of an existing runtime (e.g. delete only, pause, resume). This currently only handles "delete".